src/opnsense/mvc/app/controllers/OPNsense/Syslog/Api/ServiceController.php

```diff
@@ -0,0 +1,201 @@
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+namespace OPNsense\Syslog\Api;
+
+use \OPNsense\Base\ApiControllerBase;
+use \OPNsense\Core\Backend;
+use \OPNsense\Syslog\Syslog;
+use \OPNsense\Core\Config;
+use \Phalcon\Filter;
+
+/**
+ * Class ServiceController
+ * @package OPNsense\Syslog
+ */
+class ServiceController extends ApiControllerBase
+{
+
+    /**
+     * restart syslog service
+     * @return array
+     */
+    public function reloadAction()
+    {
+        if ($this->request->isPost()) {
+            // close session for long running action
+            $this->sessionClose();
+
+            $backend = new Backend();
+
+            // generate template
+            $backend->configdRun("template reload OPNsense.Syslog");
+
+            // (res)start daemon
+            $backend->configdRun("syslog stop");
+            $status = $backend->configdRun("syslog start");
+            $message = chop($status) == "OK" ? gettext("Service reloaded") : "";
+
+            return array("status" => $status, "message" => $message);
+        } else {
+            return array("status" => "failed", "message" => gettext("Wrong request"));
+        }
+    }
+
+    /**
+     * delete log files
+     * @return array
+     */
+    public function resetLogFilesAction()
+    {
+        if ($this->request->isPost()) {
+
+            $this->sessionClose();
+
+            $backend = new Backend();
+            $backend->configdRun("syslog stop");
+
+            $mdl = new Syslog();
+            $result = array();
+            $deleted = array();
+            foreach($mdl->LogTargets->Target->__items as $uuid => $target) {
+                if($target->ActionType == 'file') {
+                    $pathname = $target->Target->__toString();
+                    if(!in_array($pathname, $deleted)) {
+                        $status = $backend->configdRun("syslog clearlog {$pathname}");
+                        $result[] = array('name' => $pathname, 'status' => $status);
+                        $deleted[] = $pathname;
+                    }
+                }
+            }
+
+            $backend->configdRun("syslog start");
+            $backend->configdRun("syslog restart_dhcpd"); // restart dhcpd in legacy way. logic from legacy code, does it needed ?
+
+            return array("status" => "ok", "message" => gettext("The log files have been reset."), "details" => $result);
+        } else {
+            return array("status" => "failed", "message" => gettext("Wrong request"));
+        }
+    }
+
+    /**
+     * clear custom log
+     * @return array
+     */
+    public function clearLogAction()
+    {
+        if ($this->request->isPost()) {
+
+            $this->sessionClose();
+
+            $filter = new Filter();
+            $filter->add('logfilename', function($value){ return preg_replace("/[^0-9,a-z,A-Z,_]/", "", $value);});
```

Comment:
Regex is strange (too many commas) and if it is not case sensitive, you can use "i". See https://secure.php.net/manual/en/reference.pcre.pattern.modifiers.php